core/tests/karma.conf.ts

```diff
@@ -105,8 +106,15 @@ module.exports = function(config) {
     singleRun: true,
     customLaunchers: {
       Chrome_Travis: {
-        base: 'Chrome',
-        flags: ['--no-sandbox']
+        // Karma can only connect to ChromeHeadless when inside Docker.
+        base: isDocker ? 'ChromeHeadless' : 'Chrome',
+        // --no-sandbox and --disable-setuid-sandbox perform similar functions
```

Comment:
Could you explain what the flags actually do?

Reply:
To be completely honest, I really am only 30% knowledgeable about why these flags are needed (besides --no-sandbox). Online, on all the troubleshooting forums for Karma not starting with ChromeHeadless, almost all of the solutions state to use --disable-setuid-sandbox and --disable-web-security in conjunction with --no-sandbox. 

I'm not sure why this combination of flags allows Karma to work, and people online don't explain it. I've looked into what the flags individually do, though I can't tell why they are needed in a Dockerized environment. 

Will it be enough just to explain what the flags do, even if it still isn't clear why the particular flag is needed in this scenario?

Reply:
In that case, I suggest providing a link to an authoritative site that explains this, so that a developer can look at it too for context.

Reply:
Got it. Done!